core/src/main/java/org/apache/accumulo/core/file/rfile/GenerateSplits.java

```diff
@@ -265,7 +272,13 @@ private static String encode(boolean encode, Text text) {
         } else if (c >= 32 && c <= 126) {
           sb.append((char) c);
         } else {
-          log.debug("Dropping non printable char: \\x{}", Integer.toHexString(c));
+          String errMsg = "non printable char: \\x" + Integer.toHexString(c);
+          // Fail if non-printable characters are detected.
+          if (!humanRead) {
+            throw new UnsupportedOperationException(
+                errMsg + " detected. Must use Base64 encoded output");
+          }
+          log.info("Dropping {}", errMsg);
```

Comment:
Since the default behavior is being changed, could simplify this code and not have the human readable flag.  Just always fail and guide towards using the base64 option. 

```suggestion
          String errMsg = "non printable char: \\x" + Integer.toHexString(c);
          // Fail if non-printable characters are detected.
            throw new UnsupportedOperationException(
                errMsg + " detected. Must use Base64 encoded output");
```

Unrelated to these changes, but handing of `\` a few lines up seems a bit sketchy.

Reply:
Replaced this code with logic that if it detects a non-standard char it throws the exception.